odoo/models.py

```diff
@@ -2555,6 +2579,21 @@ def _init_column(self, column_name):
                 self._table, column_name, field.column_format, column_name)
             self._cr.execute(query, (value,))
 
+    def _init_column_related(self, field):
+        comodel = self.env[field.related_field.model_name]
+        self.env.cr.execute("""
+            UPDATE "{model_table}" AS x
+            SET "{model_field}" = y."{comodel_field}"
+            FROM "{comodel_table}" AS y
+            WHERE x."{join_field}" = y.id
+        """.format(
+            model_table=self._table,
+            model_field=field.name,
+            comodel_table=comodel._table,
+            comodel_field=field.related[1],
+            join_field=field.related[0],
+        ))
```

Comment:
runbot is read becase of this:

> Possible SQL injection risk. (E8501) at odoo/odoo/models.py:2584

Reply:
I use `psycopg.sql.SQL` for bulding such queries. Do you know how to use it?

Reply:
Quick translation (maybe not full accurate):

```python
        from psycopg2 import sql
        self.env.cr.execute(sql.SQL("""
            UPDATE {model_table} AS x
            SET {model_field} = y.{comodel_field}
            FROM {comodel_table} AS y
            WHERE x.{join_field} = y.id
        """).format(
            model_table=sql.Identifier(self._table),
            model_field=sql.Identifier(field.name),
            comodel_table=sql.Identifier(comodel._table),
            comodel_field=sql.Identifier(field.related[1]),
            join_field=sql.Identifier(field.related[0]),
        ))
```

Reply:
I did as you suggested and it works :+1: